internal/datastore/spanner/stats.go

```diff
@@ -46,57 +51,86 @@ func (sd spannerDatastore) Statistics(ctx context.Context) (datastore.Stats, err
 		return datastore.Stats{}, fmt.Errorf("unable to read namespaces: %w", err)
 	}
 
-	var estimate spanner.NullInt64
-	if err := sd.client.Single().Query(ctx, spanner.Statement{SQL: queryRelationshipEstimate}).Do(func(r *spanner.Row) error {
-		return r.Columns(&estimate)
-	}); err != nil {
-		return datastore.Stats{}, fmt.Errorf("unable to read row counts: %w", err)
-	}
-
-	return datastore.Stats{
-		UniqueID:                   uniqueID,
-		ObjectTypeStatistics:       datastore.ComputeObjectTypeStats(allNamespaces),
-		EstimatedRelationshipCount: uint64(estimate.Int64),
-	}, nil
-}
-
-func updateCounter(ctx context.Context, rwt *spanner.ReadWriteTransaction, change int64) error {
-	newValue := change
-
-	counterID := make([]byte, 2)
-	// nolint:gosec
-	// G404 use of non cryptographically secure random number generator is not concern here,
-	// as this is only used to randomly distributed the counters across multiple rows and reduce write contention
-	_, err := rand.New(rng).Read(counterID)
-	if err != nil {
-		return fmt.Errorf("unable to select random counter: %w", err)
-	}
+	// If there is not yet a cached estimated bytes per relationship, read a few relationships and then
+	// compute the average bytes per relationship.
+	sd.cachedEstimatedBytesPerRelationshipLock.RLock()
```

Comment:
do we know how often is the `Statistics` method polled? I think it's by default 1h right? Are we sure it's not called from anywhere else?

Reply:
Its not but since there aren't any guarantees it won't be, I ensured it was locked